package/MDAnalysis/topology/MOL2Parser.py

```diff
@@ -146,17 +158,35 @@ def parse(self, **kwargs):
             resids.append(resid)
             resnames.append(resname)
             charges.append(charge)
+            elements.append(atom_type.strip(".")[0])
 
         n_atoms = len(ids)
 
-        masses = guessers.guess_masses(types)
+        validated_elements = []
+        invalid_elements = set()
+        for elem in elements:
+            if elem.capitalize() in SYMB2Z:
+                validated_elements.append(elem.capitalize())
+            else:
+                invalid_elements.add(elem.capitalize())
+                validated_elements.append('')
+
+        # Print warnings about invalid elements
+        for elem in invalid_elements:
```

Comment:
I might be wrong here, but AFAIK warnings will only output one warning if encountering multiples of the same warning, i.e. your previous code and this one _should_ be equivalent in terms of number of outputs?

There's a performance cost between the two, but iirc I benchmarked this on a large PDB once and the cost wasn't noticeable (probably because there are much larger bottlenecks in the PDBReader).

Reply:
That being said, maybe it's cleaner to just have one output for all elements? (is this was @richardjgowers was saying above?))

Reply:
Do you mean printing them out as a list?

Reply:
yeah, I'm not sure how it'll affect readability, but at least you've only got one warning instead of N bad elements warnings.

Reply:
```
Unknown elements found for some atoms: {'X', 'XX'}. These have been given an empty element record.
```